test/com/google/enterprise/adaptor/DocIdSenderTest.java

```diff
@@ -437,36 +584,33 @@ public void testPushGroupsReplaceAllGroupsSingleFeed() throws Exception {
   @SuppressWarnings("unchecked")
   public void testPushGroupsReplaceAllGroupsAlternateBuffer() throws Exception {
     config.setValue("gsa.version", "7.4.0-1");
-    docIdSender = new DocIdSender(fileMaker, fileSender, fileArchiver, journal,
-        config, adaptor);
-
     assertNull(docIdSender.pushGroupDefinitions(sampleGroups(),
-        EVERYTHING_CASE_SENSITIVE, FULL, null, null));
+        EVERYTHING_CASE_SENSITIVE, FULL, "foo", null));
 
     assertEquals(2, fileMaker.i);
     assertEquals(
         expectedResult(Integer.MAX_VALUE, sampleGroups(), emptyGroups()),
         fileMaker.groupses);
     assertEquals(ImmutableList.of("incremental", "full"), fileSender.feedtypes);
+    assertEquals(ImmutableList.of("foo-FULL1", "foo"), fileSender.groupsources);
 
     assertNull(docIdSender.pushGroupDefinitions(sampleGroups(),
-        EVERYTHING_CASE_SENSITIVE, FULL, null, null));
+        EVERYTHING_CASE_SENSITIVE, FULL, "foo", null));
 
     assertEquals(4, fileMaker.i);
```

Comment:
Maybe add a comment before this group that the data collected by the mock fileSender is cumulative and includes previous results.